src/rgw/rgw_sal_motr.cc

```diff
@@ -3576,6 +3596,9 @@ int MotrStore::next_query_by_name(string idx_name,
     else
       next_key = key_out[i + k - 1] + " ";
     ldout(cctx, 0) << "do_idx_next_op(): next_key=" << next_key << dendl;
+    // Resizing keys & vals vector to the count of remaining keys.
```

Comment:
1. max-keys = 4;
2. keys_out = vals_out =  vector<type>(4)
3. Iteration 1
    * `do_idx_next_op(&idx, keys, vals)`  fetched 4 keys
    * After processing for delimiters we got 3 keys.  (i = 0, k = 3)
    * we assigned values at `vals_out[i+k] = vals_out[0/1/2]`
4. Iteration 2
   *  `do_idx_next_op(&idx, keys, vals)`  will fetch more 4 keys
   * after processing for delimiters we still have 4 keys. (i =3, k = 4)
   * Now we are trying to assign values at `vals_out[i+k] = vals_out[3/4/5]`. & these index are invalid. 
   * This results in a rgw crash.

* The Objective is not to fetch 4 more keys when we need only 1 key. 
* Since in the function `do_idx_next_op(&idx, keys, vals)` we are fetching keys/values equal to the size of these vector, that is why I resized them.
* Another way is to pass num_of_keys as an extra/optional argument but for that, we might have to do these changes at many places, that is why I preferred this approach.